Wabbajack/View Models/CompilerConfigVM.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Reactive.Linq;
+using System.Windows.Media.Effects;
+using System.Windows.Media.Imaging;
+using ReactiveUI;
+using ReactiveUI.Fody.Helpers;
+using Wabbajack.Lib;
+
+namespace Wabbajack
+{
+    public class CompilerConfigVM : ViewModel
+    {
+        private MainWindowVM _mainWindow;
+
+        private readonly ObservableAsPropertyHelper<ViewModel> _configArea;
+        public ViewModel ConfigArea => _configArea.Value;
+
+        private readonly Lazy<MO2CompilerConfigVM> _mo2CompilerConfig;
+
+        public BitmapImage MO2Image => UIUtils.BitmapImageFromResource("Wabbajack.Resources.MO2Button.png");
+        public BitmapImage VortexImage => UIUtils.BitmapImageFromResource("Wabbajack.Resources.VortexButton.png");
+
+        [Reactive]
+        public Effect MO2Effect { get; set; }
+        [Reactive]
+        public Effect VortexEffect { get; set; }
+
+        private readonly Effect StdBlur = new BlurEffect{Radius = 6};
+
+        [Reactive]
+        public bool ModManager { get; set; }
+
+        public IReactiveCommand BackCommand { get; }
+        public IReactiveCommand UseMO2Command { get; }
+        public IReactiveCommand UseVortexCommand { get; }
+
+        public CompilerConfigVM(MainWindowVM mainWindow)
+        {
+            _mainWindow = mainWindow;
+
+            _mo2CompilerConfig = new Lazy<MO2CompilerConfigVM>(() => new MO2CompilerConfigVM(this));
+
+            MO2Effect = null;
+            VortexEffect = null;
+
+            BackCommand = ReactiveCommand.Create(() => { _mainWindow.CurrentPage = Page.StartUp; });
+            UseMO2Command = ReactiveCommand.Create(() =>
+            {
+                SwapEffects(true);
+                ModManager = true;
+            });
+            UseVortexCommand = ReactiveCommand.Create(() =>
+            {
+                SwapEffects(false);
+                ModManager = false;
+            });
+
+            _configArea = this.WhenAny(x => x.ModManager).Select<bool, ViewModel>(a => a == false ? default : _mo2CompilerConfig.Value).ToProperty(this, nameof(ConfigArea));
+        }
+
+        public void Compile(string source)
+        {
+            _mainWindow.Compile(source);
+        }
+
+        // swaps the blur between the MO2 and Vortex button
+        private void SwapEffects(bool b)
+        {
+            if(MO2Effect == null && VortexEffect == null)
+                if (b)
+                    VortexEffect = StdBlur;
+                else
+                    MO2Effect = StdBlur;
+            else
+            {
+                if (MO2Effect == StdBlur && b)
+                {
+                    MO2Effect = null;
+                    VortexEffect = StdBlur;
+                }
+                else if (VortexEffect == StdBlur && !b)
+                {
+                    VortexEffect = null;
+                    MO2Effect = StdBlur;
+                }
+            }
+        }
```

Comment:
This sort of logic should preferably be powered by the View.  VMs should know as little as possible about the actual end display as they can.  For all they know, 10 different views might be bound to the same VM and be displaying different things.  

As an example, in a game, perhaps, you might have an NpcVM, with health, name, inventory, whatever.  An InventoryView might bind to an NPC, and display its inventory by binding to it, while a separate HealthBarView might also bind to the same object, and display the health.   NpcVM should be unaware that InventoryView/HealthBarView exist, or even care.  100 things might be bound to it and be using some of its data, for all it knows.

So back to our current code here:  The active effect should be managed by some XAML style triggers.  They should bind to some enum "Vortex/MO2", and if vortex, swap to one effect vs if MO2 swap to another. etc.  The fact that an effect even exists or is something that needs to be swapped is very specific to the view itself, and should stay in that realm

Reply:
The ModManager enum is inside my `vortex-support` PR, will make sense once this is merged and I continue with `vortex-support`. 
I did try to use triggers and effects inside the XAML and failed miserably but can try again.